addon/components/async-tree.js

```diff
@@ -29,7 +29,7 @@ export default Component.extend(ResizeAware, {
         return this.get('initial') !== this.get('_initial') ||
                this.get('initial-filter') !== this.get('_initial-filter');
       }
-  }),
+  }).volatile(),
```

Comment:
Everything else is good, but this is not necessary. If you want to reset trigger this code than just give it a new array instead of mutating the array.